src/components/portrait/portrait.test.js

```diff
@@ -0,0 +1,262 @@
+    );
+
+    it.each(testData)(
+      "should check tooltipId as %s for Portrait component",
+      (tooltipId) => {
+        CypressMountWithProviders(<PortraitComponent tooltipId={tooltipId} />);
+        portraitPreview().realHover();
+        getDataElementByValue("tooltip")
+          .should("be.visible")
+          .and("have.id", tooltipId);
+      }
+    );
+
+    it.each([true, false])(
+      "should check when tooltip visibility is %s for Portrait component",
+      (boolValue) => {
+        CypressMountWithProviders(
+          <PortraitComponent tooltipIsVisible={boolValue} />
+        );
+        if (boolValue === true) {
+          getDataElementByValue("tooltip")
+            .should("be.visible")
+            .and("have.text", "Rebecca Smith");
+        } else {
+          getDataElementByValue("tooltip").should("not.exist");
+        }
+      }
+    );
+
+    it.each(["top", "bottom", "left", "right"])(
+      "should render Portrait with the tooltip in the %s position",
+      (tooltipPosition) => {
+        CypressMountWithProviders(
+          <Box width="700px" height="108px">
+            <div
+              style={{
+                padding: "200px",
+              }}
+            >
+              <PortraitComponent tooltipPosition={tooltipPosition} />
+            </div>
+          </Box>
+        );
+        getDataElementByValue("tooltip")
+          .should("be.visible")
+          .and("have.attr", "data-placement", tooltipPosition);
+      }
+    );
+
+    it("should check error tooltipType for Portrait component", () => {
+      CypressMountWithProviders(<PortraitComponent tooltipType="error" />);
+      getDataElementByValue("tooltip").and(
+        "have.css",
+        "background-color",
+        "rgb(203, 55, 74)"
+      );
+    });
+
+    it.each([
+      ["medium", "14px"],
+      ["large", "16px"],
+    ])(
+      "should render Portrait with the tooltip in the %s size",
+      (tooltipSize, fontSize) => {
+        CypressMountWithProviders(
+          <PortraitComponent tooltipSize={tooltipSize} />
+        );
+        getDataElementByValue("tooltip")
+          .should("be.visible")
+          .and("have.css", "font-size", fontSize);
+      }
+    );
+
+    it.each(colors)(
+      "should check tooltip background-color as %s for Portrait component",
+      (name, color) => {
+        CypressMountWithProviders(<PortraitComponent tooltipBgColor={color} />);
+        getDataElementByValue("tooltip").should(
+          "have.css",
+          "background-color",
+          color
+        );
+      }
+    );
+
+    it.each(colors)(
+      "should check tooltip font color as %s for Portrait component",
+      (name, color) => {
+        CypressMountWithProviders(
+          <PortraitComponent tooltipFontColor={color} isFocused />
+        );
+        getDataElementByValue("tooltip").should("have.css", "color", color);
+      }
+    );
+
+    describe("should render Portrait component for event tests", () => {
```

Comment:
This `describe` block should be outside of the prop tests describe as it is testing action events not props. Currently, it's sitting inside of that, as though it is a sub-set of the props block.